### PR TITLE
ci: Enable the gui in the tsan build

### PR DESCRIPTION
**Pull request by hebasto**
This PR is a reincarnation of #19162.

### CODE REVIEW

**Comment by practicalswift** [comment]
Strong Concept ACK

File: depends/packages/qt.mk [hunk 1]
```diff
@@ -140,7 +140,11 @@ $(package)_config_opts_linux += -no-feature-vulkan
 $(package)_config_opts_linux += -dbus-runtime
 $(package)_config_opts_arm_linux += -platform linux-g++ -xplatform bitcoin-linux-g++
 $(package)_config_opts_i686_linux  = -xplatform linux-g++-32
+ifneq (,$(findstring clang++ -stdlib=libc++,$($(1)_cxx)))
```
[the remark this answers is not in the document]
Comment (hebasto) [reply depends/packages/qt.mk]: Suggesting to continue this discussion in #22814 where this commit belongs to.

**Comment by fanquake** [comment]
Please rebase this on master.

**Comment by hebasto** [comment]
> Please rebase this on master.

Rebased.

**Comment by MarcoFalke** [comment]
review ACK ec7b7d4a36e6e7dfa708f64666f96cd97f83f6dd

**Pull request merged**